From: Simon McVittie <smcv@collabora.com>
Date: Wed, 3 Dec 2025 15:37:28 +0000
Subject: testver: Assert that SDL_GetRevision() starts with SDL-

Signed-off-by: Simon McVittie <smcv@collabora.com>
(cherry picked from commit 65ef4df6d4d4afae3746df934837852f84345564)
Bug: https://github.com/libsdl-org/SDL/issues/14575
Applied-upstream, 3.2.30, commit:ddb497e81596a577d9822c8e1c394fb80e22f29d
---
 test/testver.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/test/testver.c b/test/testver.c
index b873038..df71f5c 100644
--- a/test/testver.c
+++ b/test/testver.c
@@ -33,9 +33,18 @@ int main(int argc, char *argv[])
             SDL_MAJOR_VERSION, SDL_MINOR_VERSION, SDL_MICRO_VERSION,
             SDL_REVISION);
     int version = SDL_GetVersion();
+    const char *revision = SDL_GetRevision();
     SDL_Log("Runtime version: %d.%d.%d (%s)",
             SDL_VERSIONNUM_MAJOR(version), SDL_VERSIONNUM_MINOR(version), SDL_VERSIONNUM_MICRO(version),
-            SDL_GetRevision());
+            revision);
+    /* This is not an API guarantee, but we try to make sure it's true,
+     * so that the version number can be extracted while debugging with:
+     * strings libSDL3.so.0 | grep SDL-
+     */
+    if (SDL_strncmp(revision, "SDL-", 4) != 0) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Revision should start with SDL-");
+        return 1;
+    }
     SDL_Quit();
     return 0;
 }
